Android/UnOHealthApp/app/src/main/java/de/mouroum/uno_health_app/MainActivity.kt

```
@@ -1,29 +1,20 @@
 package de.mouroum.uno_health_app
 
-import android.content.Intent
 import android.os.Bundle
-import android.util.Log
 import android.view.View
 import android.widget.Toast
-import androidx.annotation.MainThread
-import androidx.core.view.isVisible
-import com.google.gson.Gson
-import de.mouroum.uno_health_app.UONApp.Companion.HOST
-import de.mouroum.uno_health_app.UONApp.Companion.MEDIA_TYPE_JSON
+import androidx.lifecycle.lifecycleScope
+import de.mouroum.uno_health_app.Prefs.Companion.KEY_CURRENT_QUESTION_ID
+import de.mouroum.uno_health_app.Prefs.Companion.KEY_CURRENT_SURVEY_ID
 import de.mouroum.uno_health_app.databinding.QuestionContainerBinding
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.RequestBody.Companion.toRequestBody
-import java.lang.Thread.sleep
-import java.net.URL
-import kotlin.concurrent.thread
+import kotlinx.coroutines.launch
 
 class MainActivity : BaseActivity() {
 
     private val adapter = MainAdapter()
 
     private val currentSurvey by lazy {
-        intent.extras!!.get("survey") as Survey
+        intent.extras?.get("survey") as Survey
```

shouldn't currentSurvey be of type Survey? for the case in which the intent does not contain a survey
also defining a constant for the magic string until we change the implementation

---

in case the survey is null, should I redirect him to the activity that displays the surveys to pick one?

---

@ciprig ?

---

Let it crash MainActivity should be called only with a survey